### ISSUE TITLE
rpc endpoints setting and rpc endpoint selection

### ISSUE THREAD

**Issue by josephlabeleven** [labeleven-dev/bettercallsol#46]
### What happened?

* rpc endpoint url is not updating when adding custom rpc

* multiple selected rpc endpoint in dropdown if they have the same urls
![image](https://user-images.githubusercontent.com/111264560/185291211-3750a356-ea52-46c0-8c87-04fae4454a7b.png)
![image](https://user-images.githubusercontent.com/111264560/185291235-6b1c55f9-e67b-4077-9a0c-cf9404577274.png)

* selected rpc endpoint is not updating if user change the rpc setting without re-selecting

### App Version

v1.0.0-alpha.38

### Web Browser

Chrome

**Comment by josephlabeleven** [comment]
* rpc endpoint url is not updating when adding custom rpc 
```
           setNotValidatedUrl(e.target.value);
            setUrl(notValidatedUrl);
```
notValidatedUrl won't get updated until the parent function finishes

**Comment by josephlabeleven** [comment]
* multiple selected rpc endpoint in dropdown if they have the same urls

Use id as the reference for the check mark;
Use the first rpc endpoint as the initial value if user has the rpc endpoints in the local storage;

**Comment by josephlabeleven** [comment]
* selected rpc endpoint is not updating if user change the rpc setting without re-selecting

add a listener to the change of rpc endpoints setting and update the selected rpc endpoint correspondingly

**Comment by josephlabeleven** [comment]
* rpc endpoint url is not updating when adding custom rpc

* multiple selected rpc endpoint in dropdown if they have the same urls

These two are closed by #58

**Comment by josephlabeleven** [comment]
* selected rpc endpoint is not updating if user change the rpc setting without re-selecting

resolved by #66

**Comment by josephlabeleven** [comment]
* selected rpc endpoint is not updating if user change the rpc setting without re-selecting
resolved by #67
